src/main/java/games/strategy/net/ClientMessenger.java

```diff
@@ -37,39 +42,42 @@
   private volatile boolean m_shutDown = false;
 
   /**
-   * Note, the name paramater passed in here may not match the name of the
+   * Note, the name parameter passed in here may not match the name of the
    * ClientMessenger after it has been constructed.
    */
   public ClientMessenger(final String host, final int port, final String name, final String mac,
       final IConnectionLogin login) throws IOException {
     this(host, port, name, mac, new DefaultObjectStreamFactory(), login);
   }
 
+
   /**
-   * Note, the name paramater passed in here may not match the name of the
+   * Note, the name parameter passed in here may not match the name of the
    * ClientMessenger after it has been constructed.
    */
+  @VisibleForTesting
   public ClientMessenger(final String host, final int port, final String name, final String mac)
       throws IOException {
-    this(host, port, name, mac, new DefaultObjectStreamFactory());
+    this(host, port, name, mac, new DefaultObjectStreamFactory(), null);
   }
 
   /**
-   * Note, the name paramater passed in here may not match the name of the
-   * ClientMessenger after it has been constructed.
-   */
-  public ClientMessenger(final String host, final int port, final String name, final String mac,
-      final IObjectStreamFactory streamFact) throws IOException {
-    this(host, port, name, mac, streamFact, null);
-  }
-
-  /**
-   * Note, the name paramater passed in here may not match the name of the
+   * Note, the name parameter passed in here may not match the name of the
    * ClientMessenger after it has been constructed.
    */
   public ClientMessenger(final String host, final int port, final String name, final String mac,
       final IObjectStreamFactory streamFact, final IConnectionLogin login)
       throws IOException {
+    Preconditions.checkNotNull(mac);
+
+    Preconditions.checkState(mac.length() == 28,
```

Comment:
Since `mac` is a method parameter, would it be more appropriate to use `Preconditions#checkArgument()` instead of `checkState()`?  This comment applies to all three occurrences of `checkState()` in this method.